tools/ea/Ea/Genesis.hs

```diff
@@ -47,89 +55,66 @@ module Ea.Genesis
 
 
 import Control.Lens
+import Control.Monad
 
 import Data.Text
+import Data.Word
 
 import Chainweb.Graph
 import Chainweb.Test.TestVersions
 import Chainweb.Version
 import Chainweb.Version.Development
+import Chainweb.Version.FastDevelopment
 import Chainweb.Version.Mainnet
 import Chainweb.Version.Testnet
 
 
 -- ---------------------------------------------------------------------- --
 -- Genesis Tx Data
 
--- | Type-safe representation of chain ids
--- for a 10-chain graph.
---
-data GChainId
-    = Zero
-    | One
-    | Two
-    | Three
-    | Four
-    | Five
-    | Six
-    | Seven
-    | Eight
-    | Nine
-    | N
-    | KAD
-    deriving (Eq, Ord, Enum)
-
-instance Show GChainId where
-  show = \case
-    Zero -> "0"
-    One -> "1"
-    Two -> "2"
-    Three -> "3"
-    Four -> "4"
-    Five -> "5"
-    Six -> "6"
-    Seven -> "7"
-    Eight -> "8"
-    Nine -> "9"
-    N -> "N"
-    KAD -> "KAD"
+-- A range of chain IDs [l,r]
+data ChainIdRange
+    = UnsafeChainIdRange !Word32 !Word32
+    deriving (Eq, Ord)
 
--- | Genesis transaction record
---
-data Genesis = Genesis
-  { _version :: ChainwebVersion
-    -- ^ chainweb version (e.g. Testnet04)
-  , _tag :: Text
-    -- ^ Module name tag
-  , _txChainId :: GChainId
-    -- ^ chain id
-  , _coinbase :: Maybe FilePath
-    -- ^ filepath to coinbase yaml
-  , _keysets :: Maybe FilePath
-    -- ^ filepath to keyset yaml
-  , _allocations :: Maybe FilePath
-    -- ^ filepath to allocation yaml
-  , _namespaces :: Maybe FilePath
-    -- ^ filepath to namespace yaml
-  } deriving (Eq, Ord, Show)
-
--- | A 'Lens'' into the transaction chain id
--- of a genesis payload
---
-txChainId :: Lens' Genesis GChainId
-txChainId = lens _txChainId (\t b -> t { _txChainId = b })
+{-# COMPLETE ChainIdRange #-}
+pattern ChainIdRange :: Word32 -> Word32 -> ChainIdRange
+pattern ChainIdRange l r <- UnsafeChainIdRange l r
```

Comment:
what is the point of this pattern synonym if it doesn't use `mkChainIdRange`?

Reply:
It should be unidirectional, so just for destructuring.